modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/direct/TransportDeleteDatabaseConfigurationAction.java

```diff
@@ -0,0 +1,132 @@
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.cluster.service.MasterServiceTaskQueue;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.ingest.geoip.IngestGeoIpMetadata;
+import org.elasticsearch.ingest.geoip.direct.DeleteDatabaseConfigurationAction.Request;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransportDeleteDatabaseConfigurationAction extends TransportMasterNodeAction<Request, AcknowledgedResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportDeleteDatabaseConfigurationAction.class);
+
+    private static final SimpleBatchedExecutor<DeleteDatabaseConfigurationTask, Void> DELETE_TASK_EXECUTOR = new SimpleBatchedExecutor<>() {
+        @Override
+        public Tuple<ClusterState, Void> executeTask(DeleteDatabaseConfigurationTask task, ClusterState clusterState) throws Exception {
+            return Tuple.tuple(task.execute(clusterState), null);
+        }
+
+        @Override
+        public void taskSucceeded(DeleteDatabaseConfigurationTask task, Void unused) {
+            logger.trace("Updated cluster state for deletion of database configuration [{}]", task.databaseId);
+            task.listener.onResponse(AcknowledgedResponse.TRUE);
+        }
+    };
+
+    private final MasterServiceTaskQueue<DeleteDatabaseConfigurationTask> deleteDatabaseConfigurationTaskQueue;
+
+    @Inject
+    public TransportDeleteDatabaseConfigurationAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            DeleteDatabaseConfigurationAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            Request::new,
+            indexNameExpressionResolver,
+            AcknowledgedResponse::readFrom,
+            EsExecutors.DIRECT_EXECUTOR_SERVICE
+        );
+        this.deleteDatabaseConfigurationTaskQueue = clusterService.createTaskQueue(
+            "delete-geoip-database-configuration-state-update",
+            Priority.NORMAL,
+            DELETE_TASK_EXECUTOR
+        );
+    }
+
+    @Override
+    protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<AcknowledgedResponse> listener)
+        throws Exception {
+        final String id = request.getDatabaseId();
+        deleteDatabaseConfigurationTaskQueue.submitTask(
```

Comment:
We rewrote deletes a bit in 2b5637115c4 -- I'm going to mark this comment resolved, but I'm going to reference the same commit on the other comment and **not** resolve it because I would like you to review the new logic.